File: math/exponentiation.py
from sys import stdin
mod = 10 ** 9 + 7


def power(a, b):
    res = 1

    while b:
        if b & 1:
            res = (res % mod * a % mod) % mod
            b -= 1
        else:
            a = (a % mod * a % mod) % mod
            b = b // 2
    return res


t = int(stdin.readline())
for _ in range(t):
    a, b = map(int, stdin.readline().split())
    print(power(a, b))

File: math/exponentiation.cc
#include <bits/stdc++.h>
using namespace std;
#define ll long long int

const int mod = 1000000007;

ll power(ll a, ll b)
{
    ll res = 1;
    while (b)
    {
        if (b & 1)
        {
            res = (res % mod * a % mod) % mod;
            b -= 1;
        }
        else
        {
            a = (a % mod * a % mod) % mod;
            b = b / 2;
        }
    }
    return res;
}

int main()
{
    int n;
    cin >> n;
    while (n--)
    {
        ll a, b;
        cin >> a >> b;
        cout << power(a, b) << '\n';
    }
    return 0;
}